### ISSUE TITLE
Create GCP package

### ISSUE THREAD

**Issue by masci** [elastic/integrations#490]
# Integration release checklist

This checklist is intended for integrations maintainers to ensure consistency
when creating or updating a Package, Module or Dataset for an Integration.

### All changes

- [ ] Change follows [development guidelines](https://github.com/elastic/integrations/tree/master/doc/development/guidelines)
- [ ] Supported versions of the monitoring target are documented
- [ ] Supported operating systems are documented (if applicable)
- [ ] Integration or [System tests](https://github.com/elastic/elastic-package/blob/master/docs/howto/system_testing.md) exist
- [ ] Documentation exists
- [ ] Fields follow [ECS](https://github.com/elastic/ecs) and [naming conventions](https://www.elastic.co/guide/en/beats/devguide/master/event-conventions.html)
- [ ] At least a manual test with ES / Kibana / Agent has been performed.
- [ ] Required Kibana version set to:

### New Package

- [ ] Screenshot of the "Add Integration" page on Fleet added

### Dashboards changes

- [ ] Dashboards exists
- [ ] Screenshots added or updated
- [ ] Datastream filters added to visualizations

### Log dataset changes

- [ ] [Pipeline tests](https://github.com/elastic/elastic-package/blob/master/docs/howto/pipeline_testing.md) exist (if applicable)
- [ ] Generated output for at least 1 log file exists

### Metric dataset changes

_This entry is currently only recommended. It will be mandatory once we provide better support for it._

- [ ] Sample event (`sample_event.json`) exists

### Filebeat module changes

- [ ] Test log files exist for the grok patterns
- [ ] Generated output for at least 1 log file exists

### Metricbeat module changes

- [ ] Example `data.json` exists and an automated way to generate it exists (`go test -data`)
- [ ] Test environment in Docker exist for integration tests

**Comment by masci** [comment]
Package creation: https://github.com/elastic/integrations/pull/459

**Comment by endorama** [comment]
@legoguy1000 thank you! I will keep an eye on it and review it.

**Comment by endorama** [comment]
Everything has being slowed down by adding tests, which required to address some limitation of our current CI setup for GCP. 

The work to run GCP tests is in progress:
- https://github.com/elastic/elastic-package/pull/603
- https://github.com/elastic/elastic-package/pull/638
- https://github.com/elastic/elastic-package/pull/665
- https://github.com/elastic/elastic-package/pull/700
- https://github.com/elastic/elastic-package/pull/701
- https://github.com/elastic/elastic-package/pull/662

**Comment by endorama** [comment]
While adding tests we stumbled upon an issue with Elastic Agent parsing it's configuration that is affecting how we perform tests: https://github.com/elastic/beats/issues/30710

**Comment by endorama** [comment]
GCP test package is waiting on the final review but we were able to overcome the issues of the past days. I expect it to be merged beginning of next week

**Comment by endorama** [comment]
GCP test package merged today! This unblocks working on multiple data streams (probably all but `billing`, for which https://github.com/elastic/elastic-package/pull/662 is required)

**Comment by legoguy1000** [comment]
@endorama we could to keep merging the individual Gcp metric sets?

**Comment by endorama** [comment]
No, unfortunately tests are still not working in this repo. We are migrating the configuration but facing current system tests for logs failing in https://github.com/elastic/integrations/pull/2704

**Comment by endorama** [comment]
While integrating tests in https://github.com/elastic/integrations/pull/2704 we discovered errors in the CI pipeline. As the changes are unrelated to #2704, I'm pausing the work on that one to fix the CI in https://github.com/elastic/integrations/pull/2707/

**Comment by endorama** [comment]
Work has been done in a [separate branch](https://github.com/endorama/integrations/tree/gcp-metrics-II) to pinpoint the issue, and we have been able to find and solve the issue.

During testing of #2707 it has been [highlighted](https://github.com/elastic/integrations/pull/2707#discussion_r832592613) that the changes introduce a possible breaking change in the configuration that we should address.
I performed some tests and I can confirm there is a breaking change that we have to address. Further details in a separate comment.

**Comment by endorama** [comment]
In #2987 we decided to move forward by creating a separate integration package called `gcp_metrics`. I'll rework implementation PR to use the new package.

**Comment by legoguy1000** [comment]
@endorama let me know when the new package is in the `gcp_metrics` branch and I can update all my other branches with the remaining data streams and we can finish this.

**Comment by endorama** [comment]
The decision of moving forward has been reverted, in favour of a different approach: an issue has been opened on Kibana (https://github.com/elastic/kibana/issues/131251), that has been closed yesterday. This issue solves the upgrade issue I found when upgrading GCP policy with a breaking change. 
The next step is to ensure the upgrade works as expected and then continue migrating the metricsets as previously decided.

**Comment by endorama** [comment]
#2707 has been updated to include latest changes and marked ready again, after positive decision in moving forward with the current implementation.

**Comment by legoguy1000** [comment]
> @legoguy1000 I'm sorry that all linked PRs have been closed! I'm not sure why it happened but was not intentional. We are testing the 2.0.0 package and we are going to work on other metricsets soon after!

No problem.  When the branch was merged and deleted, any PR targeting it gets closed. I had to rebase it off of main anyways.
